src/Functions/generateSnowflakeID.cpp

```diff
@@ -165,24 +166,41 @@ class FunctionGenerateSnowflakeID : public IFunction
     {
         FunctionArgumentDescriptors mandatory_args;
         FunctionArgumentDescriptors optional_args{
-            {"expr", nullptr, nullptr, "Arbitrary expression"}
+            {"expr", nullptr, nullptr, "Arbitrary expression"},
+            {"machine_id", static_cast<FunctionArgumentDescriptor::TypeValidator>(&isNativeUInt), nullptr, "Optional machine ID in UInt*"}
         };
         validateFunctionArguments(*this, arguments, mandatory_args, optional_args);
 
         return std::make_shared<DataTypeUInt64>();
     }
 
-    ColumnPtr executeImpl(const ColumnsWithTypeAndName & /*arguments*/, const DataTypePtr &, size_t input_rows_count) const override
+    ColumnPtr executeImpl(const ColumnsWithTypeAndName & arguments, const DataTypePtr &, size_t input_rows_count) const override
     {
         auto col_res = ColumnVector<UInt64>::create();
         typename ColumnVector<UInt64>::Container & vec_to = col_res->getData();
+        vec_to.resize(input_rows_count);
 
-        if (input_rows_count != 0)
+        uint64_t machine_id = 0;
+        if (arguments.size() == 2 && input_rows_count > 0)
         {
-            vec_to.resize(input_rows_count);
+            const auto & column = arguments[1].column;
+            if (column && !column->empty())
+                machine_id = column->getUInt(0);
+        }
+
+        if (machine_id == 0)
+            machine_id = getMachineId();
+
+        /// Ensure machine_id is within the valid range
```

Comment:
Please change the comment to `Truncate machine id to 10 bits`

Reply:
Btw., it seems that `getMachineId()` already returns 10 bits, so the truncation in l. 194 is technically not necessary. We could do it only if the user provided a machine id.